server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java

```diff
@@ -57,4 +327,363 @@ public void testSimpleTerminateAfterCount() throws Exception {
         assertHitCount(searchResponse, max);
         assertFalse(searchResponse.isTerminatedEarly());
     }
+
+    public void testSimpleTerminateAfterTrackTotalHitsUpTo() throws Exception {
```

Comment:
Added new test here to try different terminate_after / track_total_hits / size edge cases